File: HerancaPoo/src/main/kotlin/classesabstratas/Cachorro.kt
package classesabstratas

class Cachorro(
    especie: String, nomeCientifico: String,
    habitat: String, ehCarnivoro: Boolean, sexo: String, idade: Int,
    val nome: String
) : Animal(especie, nomeCientifico, habitat, ehCarnivoro, sexo, idade){

    override fun correr() {
        println("Correndo como um dog")
    }

    override fun dormir() {
        println("Dormindo como um dog")
    }

    override fun andar() {
        println("Andando como um dog")
    }

    override fun comer() {
        println("Comendo como um dog")
    }

    override fun emitirSom() {
        println("Auau")
    }

    fun abanarRabo(){
        println("Abanando o Rabo")
    }

    fun rosnar(){
        println("Rosnar")
    }

}

File: Javascript/CorrecaoHeranca/src/cachorro.ts
import { Animal } from "./animal";

export class Cachorro implements Animal{

    nome: string
    idade: number

    constructor(
        nome: string,
        idade: number
    ){
        this.nome = nome
        this.idade = idade
    }

    emitirSom(): void {
        console.log("Auau")
    }

    locomocao(): void {
        console.log("Correndo")
    }
    
}